src/mechanics/main_menu_buttons.rs

```diff
@@ -0,0 +1,27 @@
+use bevy::prelude::*;
+use bevy::app::AppExit;
+
+use crate::AppState;
+use crate::visuals::main_menu::MainMenuButtonTypes;
+
+pub fn button_system(
```

Comment:
Would you mind making a unit test that checks whether the state has transitioned properly when the Play button is clicked, checking to see that all of the MainMenuUI items are de-spawned as they should be as well?

Reply:
Done